WcaOnRails/spec/controllers/team_members_controller_spec.rb

```diff
@@ -0,0 +1,164 @@
+# frozen_string_literal: true
+require 'rails_helper'
+
+describe TeamMembersController do
+  let(:team_member_to_create) { FactoryGirl.build(:team_member) }
+  let!(:team_member) { FactoryGirl.create(:team_member) }
+
+  describe "GET #new" do
+    context "when not signed in" do
+      sign_out
+
+      it 'redirects to the sign in page' do
+        get :new, committee_id: team_member.team.committee.slug, team_id: team_member.team.slug
+        expect(response).to redirect_to new_user_session_path
+      end
+    end
+
+    context "when signed in as admin" do
+      sign_in { FactoryGirl.create(:admin) }
+
+      it 'shows the team_member new page' do
+        get :new, committee_id: team_member.team.committee.slug, team_id: team_member.team.slug
+        expect(response).to render_template :new
+      end
+    end
+
+    context 'when signed in as a demoted admin' do
+      sign_in { FactoryGirl.create :admin_demoted }
+
+      it 'does not allow access' do
+        get :new, committee_id: team_member.team.committee.slug, team_id: team_member.team.slug
+        expect(response).to redirect_to root_url
+      end
+    end
+
+    context 'when signed in as a regular user' do
+      sign_in { FactoryGirl.create :user }
+
+      it 'does not allow access' do
+        get :new, committee_id: team_member.team.committee.slug, team_id: team_member.team.slug
+        expect(response).to redirect_to root_url
+      end
+    end
+  end
+
+  describe 'POST #create' do
+    context 'when not signed in' do
+      it 'redirects to the sign in page' do
+        post :create, committee_id: team_member_to_create.committee.slug, \
+                      team_id: team_member_to_create.team.slug, \
+                      team_member: { user_id: team_member_to_create.user_id, \
```

Comment:
This is really redundant you can use `team_member: team_member_to_create.attributes` I hope.